src/WebJobs.Script.WebHost/ContainerManagement/LinuxContainerInitializationHostService.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.ContainerManagement
+{
+    public class LinuxContainerInitializationHostService : IHostedService
+    {
+        private readonly ILinuxContainerInitializationService _linuxContainerInitializationService;
+        private readonly ILogger _logger;
+
+        public LinuxContainerInitializationHostService(ILinuxContainerInitializationService linuxContainerInitializationService, ILoggerFactory loggerFactory)
+        {
+            _linuxContainerInitializationService = linuxContainerInitializationService;
+            _logger = loggerFactory.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
+        }
+
+        public async Task StartAsync(CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Initializing LinuxContainerInitializationService.");
+            await _linuxContainerInitializationService.Run(cancellationToken);
```

Comment:
The context reader is only needed within this host service right? Then I think you can simplify by getting rid of the ILinuxContainerInitializationService completely and just add all the logic/methods to this class?

Reply:
Yes. I could, but keeping it as separate class makes it easy to mock it.  Let me know if you prefer the virtual method approach.

Reply:
I don't see you mocking ILinuxContainerInitializationService anywhere - looks like your tests create a concrete instance. Generally however I agree with you that enabling service mocking is a good reason for factoring out into separate types.

Note that you can mock using virtuals, without defining interfaces. I.e. the IHostAssignmentContextReader/ILinuxContainerInitializationService/etc. interfaces aren't really necessary. Interfaces are useful if you expect to have different implementations in the future, but that isn't the case for any of these.